### ISSUE TITLE
[Programmatic Access - Make code - Share project]: 'Project link' read-only form field does not have a label.

### ISSUE THREAD

**Issue by vroopea** [microsoft/pxt-microbit#5472]
### User Experience:
Visually impaired people who depend on screen readers for navigation will get impacted as they won't be able to know the purpose of the read-only field and cannot copy the value for the project link.

### Note:
User credentials should NOT be included in the bug.

### Repro Steps:

1. Launch application URL : https://makecode.microbit.org/.
2. Home page will open.
3. Navigate "New Project" option and select it.
4. Editor screen will open.
5. Navigate to "Share" option appearing in header section and select it.
6. Navigate through all the controls of "Share Project" dialog and invoke share project.
7. Turn on the AI (Accessibility Insights) and observe whether project link read-only field has label or not.

### Actual Result:
Issue: 'Project link' read-only form field does not have a label.

**Observation:**
When screen reader focus lands on the 'Project link' read-only field, the screen reader announces as "https://makecode.microbit.org/_1fE2E93pgaU6 read-only".

### Expected Result:

- Ensures every form element has a label
- Accessible name should be defined 'Project link' read-only form field does not have a label.

**Example:** Aria-label="Project link" should be defined for the read-only field.

**Note:** The same issue is observed throughout the application for the similar controls.

### Issue: 
Ensures every form element has a label (label - https://accessibilityinsights.io/info-examples/web/label)

### Target application: 
Microsoft MakeCode for micro:bit - https://makecode.microbit.org/#editor

### Element path: 
.common-input

### Snippet: 
`<input class="common-input" role="textbox" tabindex="0" type="text" readonly="" autocomplete="off" autocorrect="off" autocapitalize="off" value="https://makecode.microbit.org/_1fE2E93pgaU6">`

### How to fix: 
**Fix any of the following:**

- Form element does not have an implicit (wrapped) <label>
- Form element does not have an explicit <label>
- aria-label attribute does not exist or is empty
- aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty
- Element has no title attribute
- Element has no placeholder attribute
- Element's default semantics were not overridden with role="none" or role="presentation".

This accessibility issue was found using Accessibility Insights for Web 2.41.0 (axe-core 4.7.2), a tool that helps find and fix accessibility issues. Get more information & download this tool at http://aka.ms/AccessibilityInsights.

### MAS Reference:
[MAS 4.1.2 – Name, Role, Value](https://aka.ms/MAS4.1.2)

### ARIA Authoring Ref Link/Fluent Patterns Guidance /HITS Guidance:

[label](https://accessibilityinsights.io/info-examples/web/label).

### UIA/DOM details available:

Yes (Refer the attachment)

### Reference Links:

- [Accessibility Insights!](https://accessibilityinsights.io/) - Identify accessibility bugs before check-in and make bug fixing faster and easier.
- External Bug Process: If this bug belongs to an external team, mark it as resolved/done and assign it back to the tester with notes on where to file/route the bug. For more information, please use this [External Bug Process (sharepoint.com)](https://microsoft.sharepoint.com/teams/cea11y/SitePages/External-Bug-Process.aspx?xsdata=MDN8MDF8fDE1MThmM2JlZGFjMDRlZDhiZmViYTUxMDA2M2NiZWI3fDcyZjk4OGJmODZmMTQxYWY5MWFiMmQ3Y2QwMTFkYjQ3fDF8MHw2Mzc3NDEyNTM2MTEzNzIwNzR8R29vZHxWR1ZoYlhOVFpXTjFjbWwwZVZObGNuWnBZMlY4ZXlKV0lqb2lNQzR3TGpBd01EQWlMQ0pRSWpvaUlpd2lRVTRpT2lJaUxDSlhWQ0k2TVRKOQ%3D%3D&sdata=MnlpSXlVMnpGT3cvZ0xJNnFWUWZWTGt6dmo1VkJMZFpIZHBvUUduNDhyMD0%3D&ovuser=72f988bf-86f1-41af-91ab-2d7cd011db47%2cv-kgeetala%40microsoft.com). “HCL Staff should not log any third-party external bugs. Should be routed to EDAD team”.
- Please reach out to [C&AI Teams channel](https://apc01.safelinks.protection.outlook.com/?url=https://teams.microsoft.com/l/team/19%253a3cd071821b694f11b054d878453cd75f%2540thread.skype/conversations?groupId%3D30e3f968-15dd-4da2-bd8e-85f27e4af2c3%26tenantId%3D72f988bf-86f1-41af-91ab-2d7cd011db47&data=02%7c01%7cAklavyaM%40hcl.com%7c24e89c6ac26f42cfe40c08d7667e49f3%7c189de737c93a4f5a8b686f4ca9941912%7c0%7c0%7c637090564906844600&sdata=22JjBMoA1ID/KnYqFa/OB%2BrkTaLh5YYsGBd/7Bm7u5w%3D&reserved=0) for any process-related queries.

### Test Environment:

- OS Version: Windows 11 23H2 (OS Build:22631.3007)
- Browser: Chrome Version 121.0.2277.4 (Official build) dev (64-bit).
- Test Matrix: Chrome + JAWS
- URL: https://makecode.microbit.org/.
<img width="1356" alt="MAS4 1 2_'Project link' read-only form field does not have a label" src="https://github.com/microsoft/pxt-microbit/assets/158465724/2308dde7-cd3e-4d4d-a4ea-ebc074c62c99">

**Comment by vroopea** [comment]
#Accessibility;#A11yMAS;#A11ySev2;#A11yAuto;#k4W;#MAS4.1.2;#HCL;#Benchmark;#Win11-Chrome;#E2E-HCL-MakeCode-Jan2024

**Comment by vroopea** [comment]
As checked the issue on URL: https://makecode.microbit.org/. The issue is still repro and severity is fine. #Regressed:04-08-24;InternalReviewed-Apr2024
